src/tiered/tiered_handle.c

```diff
@@ -19,27 +19,52 @@ __tiered_name_check(WT_SESSION_IMPL *session, WT_TIERED *tiered)
 {
     WT_DECL_RET;
     WT_FILE_SYSTEM *bucket_fs;
+    size_t len, obj_len;
     u_int obj_count, i;
     char **obj_files;
-    const char *name;
+    const char *name, *obj_name, *obj_uri;
 
     bucket_fs = tiered->bstorage->file_system;
     name = tiered->iface.name;
+    obj_name = obj_uri = NULL;
     WT_ASSERT(session, WT_PREFIX_MATCH(name, "tiered:"));
     WT_PREFIX_SKIP(name, "tiered:");
     /* See if this name exists in the shared storage. */
     __wt_verbose(session, WT_VERB_TIERED, "NAME_CHECK: check for %s", name);
-    WT_ERR(bucket_fs->fs_directory_list(
+    WT_RET(bucket_fs->fs_directory_list(
       bucket_fs, (WT_SESSION *)session, NULL, name, &obj_files, &obj_count));
     __wt_verbose(session, WT_VERB_TIERED, "NAME_CHECK: Got %d files", (int)obj_count);
-    for (i = 0; i < obj_count; ++i)
+    if (obj_count == 0)
+        goto done;
+    /*
+     * We need to handle overlapping naming. Generate an object name so that we know the maximum
+     * length that our name should be and we can check for the right form.
+     */
+    WT_ERR(__wt_tiered_name(session, &tiered->iface, 1, WT_TIERED_NAME_OBJECT, &obj_uri));
+    obj_name = obj_uri;
```

Comment:
Do we need both `obj_name` and `obj_uri`?  It looks like you could just use one throughout this code.

Reply:
The full URI is returned by `wt_tiered_name` in `obj_uri`. I want to use just the name portion so `obj_name` strips off the leading `object:` from the URI. The file names returned by `list_directory` are just the names which is why I use `obj_name`.

Reply:
Ah yes.  My mistake.